### PR TITLE
PI-848 handle person died release message

### PR DESCRIPTION
**Pull request by anthony-britton-moj**

### CODE REVIEW

File: projects/prison-custody-status-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/release/PersonDied.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private val contactTypeRepository: ContactTypeRepository,
+    private val contactRepository: ContactRepository,
+    private val contactAlertRepository: ContactAlertRepository
+) {
+    fun inCustody(nomsId: String, dateTime: ZonedDateTime) {
+        val people = personRepository.findByNomsNumberAndSoftDeletedIsFalse(nomsId)
+        val person = when (people.size) {
+            1 -> people[0]
+            0 -> throw NotFoundException("Person", "nomsNumber", nomsId)
+            else -> throw IllegalStateException("More than one case with the same Noms Number: $nomsId")
```
Comment (marcus-bcl) [inline projects/prison-custody-status-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/release/PersonDied.kt:32]: Currently we ignore these errors for normal messaging, should we be doing the same here?
See https://github.com/ministryofjustice/hmpps-probation-integration-services/blob/871a34ba347c7df4373ec8d0417a189d5dabb0d9/projects/prison-custody-status-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/event/EventService.kt#L14-L18

File: projects/prison-custody-status-to-delius/src/test/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/release/ReleaseServiceTest.kt [hunk 2]
```diff
@@ -84,23 +87,55 @@ internal class ReleaseServiceTest {
...
+        }
+        assertThrows<IgnorableMessageException> {
+            releaseService.release(
+                "",
+                "",
+                "TRANSFERRED",
+                "BL",
+                ZonedDateTime.now()
+            )
+        }
```
Comment (marcus-bcl) [inline projects/prison-custody-status-to-delius/src/test/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/release/ReleaseServiceTest.kt:125]: Not an issue, but it seems there are a few unrelated formatting changes?

**Review by marcus-bcl — COMMENTED** [review]
Looks good, minor question about missing/duplicate NOMS number

**Review by stevomcallister — APPROVED** [review]

File: projects/prison-custody-status-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/release/PersonDied.kt [hunk 3]
```diff
@@ -28,8 +28,8 @@ class PersonDied(
         val people = personRepository.findByNomsNumberAndSoftDeletedIsFalse(nomsId)
         val person = when (people.size) {
             1 -> people[0]
-            0 -> throw NotFoundException("Person", "nomsNumber", nomsId)
-            else -> throw IllegalStateException("More than one case with the same Noms Number: $nomsId")
+            0 -> throw IgnorableMessageException("MissingNomsNumber: $nomsId")
+            else -> throw IgnorableMessageException("DuplicateNomsNumber: $nomsId")
```
Comment (marcus-bcl) [inline projects/prison-custody-status-to-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/release/PersonDied.kt:32]: I think the exception message is what gets added to app insights, so appending the $nomsId will make it harder to aggregate the events.  The noms number is added as a Telemetry property anyway so we can still see the value if we need to.

**Review by marcus-bcl — APPROVED** [review]

**Review by stevomcallister — APPROVED** [review]
